compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -509,6 +517,64 @@ encodeModuleRule =
         let (hash, bs) = SS.encodeModule lfVersion m
         return ([], Just (mconcat $ hash : map fst encodedDeps, bs))
 
+-- hlint
+
+-- We need to provide hlint the path to the directory of a file
+-- containing a base configuration file called 'hlint.yaml'.
+-- We test with one version of 'hlint.yaml' and deploy with another
+-- (at the current time, there are good reasons for this). The code
+-- below (which relies on the behavior of `package_app` deploying to
+-- '~/.daml/sdk/x.x.x/damlc/resources/daml-ide-core/data') selects one
+-- or the other depending on context.
+getHlintDataDir :: IO FilePath
+getHlintDataDir = do
+  root <- locateRunfiles $
+    mainWorkspace </> "compiler/damlc/daml-ide-core"
+  let test = root </> "data/test"
+      prod = root </> "daml-ide-core/data/prod"
+  useProd <- System.Directory.doesDirectoryExist prod
+  return $ if useProd then prod else test
+
+-- This next function harvests hlint settings using 'hlint.yaml'. What
+-- we don't have at this time but I will follow up with in a later
+-- update, is the ability to augment the settings via per-project
+-- overrides from user provided '.hlint.yaml' files (found by by
```

Comment:
I don’t think it is reasonable to ship damlc with always on hlint that cannot be configured by users. At the very least we need an option to turn it off completely before we ship this.

Reply:
Agree. I think I would rather invest time at this point in being able to turn the feature on/off than writing the logic for per-project `.hlint.yaml` files (although we do have to implement that before long). I think time is better spent right now on converting rules in hlint to ghc-lib-parser.

Reply:
Yeah just implementing an on/off switch for now is totally fine.